File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/player/PlayerResolver.java
package top.likoslupus.cellulosesz.api.player;

import org.jspecify.annotations.Nullable;
import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import java.util.UUID;
import java.util.concurrent.CompletableFuture;

public interface PlayerResolver {

    ResolvedPlayer resolveKnown(String input, @Nullable CellPlayer viewer);

    ResolvedPlayer resolveKnown(UUID uuid, @Nullable CellPlayer viewer);

    CompletableFuture<ResolvedPlayer> resolve(String input, @Nullable CellPlayer viewer);

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/player/PlayerResolver.kt
package top.likoslupus.cellulosesz.api.player

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import java.util.*
import java.util.concurrent.CompletableFuture

public interface PlayerResolver {

    public fun resolveKnown(input: String, viewer: CellPlayer?): ResolvedPlayer

    public fun resolveKnown(uuid: UUID, viewer: CellPlayer?): ResolvedPlayer

    public fun resolve(input: String, viewer: CellPlayer?): CompletableFuture<ResolvedPlayer>

}
